flask_restless/helpers.py

```diff
@@ -180,10 +180,33 @@ def is_date_field(model, fieldname):
     """Returns ``True`` if and only if the field of `model` with the specified
     name corresponds to either a :class:`datetime.date` object or a
     :class:`datetime.datetime` object.
-
     """
-    fieldtype = get_field_type(model, fieldname)
-    return isinstance(fieldtype, Date) or isinstance(fieldtype, DateTime)
+    is_datelike = lambda field_type: isinstance(field_type, Date) or \
+        isinstance(field_type, DateTime)
+    return iteratively_check_parent_is_of_field(
+        model,
+        fieldname,
+        is_datelike
+    )
+
+
+def iteratively_check_parent_is_of_field(
+    model,
+    fieldname,
+    check_field_function
+):
+    """SQLAlchemy supports `TypeDecorators` of `TypeDecorators` following
```

Comment:
Bad documentation string. Again, this is (mostly) describing the implementation, not the behavior of the function.

Something like

    """Returns ``True`` if and only if the field implementation meets the condition provided by
    `check_field_function`, regardless of `TypeDecorator` wrappers.

     `model` is ...

     `fieldname` is ....

    `check_field_function` is ...

    """